packages/paypal-commerce-integration/src/paypal-commerce-fastlane/paypal-commerce-fastlane-payment-strategy.ts

```diff
@@ -133,22 +133,33 @@ export default class PaypalCommerceFastlanePaymentStrategy implements PaymentStr
         const state = this.paymentIntegrationService.getState();
         const cartId = state.getCartOrThrow().id;
 
-        const { orderId } = await this.paypalCommerceRequestSender.createOrder(methodId, {
-            cartId,
-        });
+        const isVaultedFlow = paymentData && isVaultedInstrument(paymentData);
+
+        try {
+            const { id = '' } = isVaultedFlow ? {} : await this.getPayPalFastlanePaymentToken();
+
+            const { orderId } = await this.paypalCommerceRequestSender.createOrder(methodId, {
+                cartId,
+            });
```

Comment:
Why don't we move createOrder method call inside `prepareVaultedInstrumentPaymentPayload` and `preparePaymentPayload` to separate and simplify the flow?

In this case it is not necessary to have this line:
`const { id = '' } = isVaultedFlow ? {} : await this.getPayPalFastlanePaymentToken();`

Reply:
It looks ok to me, since both flows have own specific logic